cluster/service.go

```diff
@@ -74,22 +93,52 @@ func (c *Service) Open(ctx context.Context, db schema.Indexer) error {
 		return fmt.Errorf("open raft store: %w", err)
 	}
 
-	bs := bootstrap.NewBootstrapper(
-		c.rpcClient,
-		c.config.NodeID,
-		c.raftAddr,
-		c.config.NodeToAddressResolver,
-		c.Raft.Ready,
-	)
+	// If FQDN resolver is enabled make sure we're also using it for the bootstrapping process
+	nodeToAddressResolver := c.config.NodeToAddressResolver
+	if c.config.EnableFQDNResolver {
+		nodeToAddressResolver = resolver.NewFQDN(resolver.FQDNConfig{
+			// We don't need to specify more config as we are only using that resolver to resolve a node id to an IP
+			// overriding the default resolver using memberlist.
+			TLD: c.config.FQDNResolverTLD,
+		})
+	}
+
+	hasState, err := raft.HasExistingState(c.Raft.store.logCache, c.Raft.store.logStore, c.Raft.store.snapshotStore)
+	if err != nil {
+		return err
+	}
+	c.log.WithField("hasState", hasState).Info("raft init")
 
```

Comment:
**thought:**  i think we have to call `c.Raft.store.init()` before to load all the raft stores, and remove the call from `store.Open()` otherwise it will always be false

Reply:
Testing locally, hasState returns false on the original start but on subsequent start will return true, so I think it's working as expected.

Reply:
aah, it's called inside `Open()`, so `raft.Open()` would have been called before it. i missed that . i thought it's in New